opencog/reasoning/RuleEngine/rule-engine-src/Rule.cc

```diff
@@ -23,50 +23,116 @@
 
 #include "Rule.h"
 
-Rule::Rule(Handle rule) {
+Rule::Rule(Handle rule)
+{
 	rule_handle_ = rule;
 }
 
 
-Rule::~Rule() {
+Rule::~Rule()
+{
 
 }
 
-Handle Rule::get_handle() {
-	return rule_handle_;
-}
-
-int Rule::get_cost() {
+int Rule::get_cost()
+{
 	return cost_;
 }
 
-void Rule::set_category(string name) {
+void Rule::set_category(string name)
+{
 	category_ = name;
 }
 
-string& Rule::get_category() {
+string& Rule::get_category()
+{
 	return category_;
 }
 
-void Rule::set_name(string name) {
+void Rule::set_name(string name)
+{
 	name_ = name;
 }
 
-string Rule::get_name() {
+string Rule::get_name()
+{
 	return name_;
 }
-void Rule::set_rule_handle(Handle h) throw (exception) {
+
+void Rule::set_handle(Handle h) throw (exception)
+{
 	rule_handle_ = h;
 }
 
-void Rule::set_cost(int p) {
+Handle Rule::get_handle()
+{
+	return rule_handle_;
+}
+
+/**
+ * Get the implicant (input) of the rule defined in a BindLink.
+ *
+ * @return the Handle of the implicant
+ */
+Handle Rule::get_implicant()
+{
+	// rules must be defined in a BindLink for pattern matching
+	if (rule_handle_->getType() != BIND_LINK)
```

Comment:
I don't know... I actually think all these checking should have been moved to set_handle to stop setting a rule with incorrectly structured BindLink.  If an exception has to be thrown, it makes more sense in this case to throw at the set function rather than the get function.